src/Makefile.am

```diff
@@ -502,12 +502,14 @@ libbitcoin_common_a_SOURCES = \
   protocol.cpp \
   psbt.cpp \
   rpc/rawtransaction_util.cpp \
+  rpc/server.cpp \
   rpc/util.cpp \
   scheduler.cpp \
   script/descriptor.cpp \
   script/sign.cpp \
   script/signingprovider.cpp \
   script/standard.cpp \
+  shutdown.cpp \
```

Comment:
Can you explain what this is doing and why?

Also, the ci is failing

Reply:
Since I've added `rpc/server.cpp` to `libbitcoin_common_a_SOURCES`, and `rpc/server.cpp` in turn includes `shutdown.h` and invokes `StartShutdown()` [here],(https://github.com/bitcoin/bitcoin/blob/master/src/rpc/server.cpp#L172) I need to include `shutdown.cpp` in `libbitcoin_common_a_SOURCES`

Otherwise we'll get a compile error:
```sh
Undefined symbols for architecture x86_64:
  "StartShutdown()", referenced from:
      std::__1::__function::__func<stop()::$_3, std::__1::allocator<stop()::$_3>, UniValue (RPCHelpMan const&, JSONRPCRequest const&)>::operator()(RPCHelpMan const&, JSONRPCRequest const&) in libbitcoin_common.a(libbitcoin_common_a-server.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

Reply:
Looks like I made a mistake resolving a merge conflict when I rebased to master, which caused CI to fail.

The CI error was:
```sh
Good job! The circular dependency "policy/fees -> txmempool -> validation -> policy/fees" is no longer present.
458Please remove it from EXPECTED_CIRCULAR_DEPENDENCIES in test/lint/lint-circular-dependencies.sh
459to make sure this circular dependency is not accidentally reintroduced.
```

I had a single merge conflict in `lint-circular-dependencies.sh` where: `"policy/fees -> txmempool -> validation -> policy/fees"` needed to be removed, but I re-added it instead of removing it when resolving the merge conflict.

Fixed now: https://github.com/bitcoin/bitcoin/pull/20286/files#diff-ee6a5803b78eb12f095b391894014b5fb2b836cef29943980b3ab684861c1f2dR23